src/main/scala/org/broadinstitute/hail/expr/Fun.scala

```diff
@@ -0,0 +1,25 @@
+package org.broadinstitute.hail.expr
+
+sealed trait Fun {
+  def retType: Type
+}
+
+sealed case class UnaryFun[T, U](retType: Type, f: (T) => U) extends Fun with Serializable with ((T) => U) {
```

Comment:
I think you don't need you need sealed on the subclasses if the parent is sealed.